app/services/action_builder.rb

```diff
@@ -20,6 +20,15 @@ def build_action(extra_attrs = {})
   def previous_action
```

Comment:
Ah yeah I see what you mean with the load in the conditional. In terms of the loop, each campaign may have perhaps ten pages, and each iteration through the loop will do the id compare and then hit the ||= and go to the next iteration. It would be a lot clearer to do an explicit break. In fact, this method also got flagged by code climate. I'll do some cleanup today. Thanks!